repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py

```diff
@@ -0,0 +1,98 @@
+from os import path, listdir
+
+from leapp.libraries.stdlib import CalledProcessError, api, run
+from leapp.models import SapHanaInfo, SapHanaInstanceInfo
+
+HANA_BASE_PATH = '/hana/shared'
+HANA_MANIFEST_PATH = 'exe/linuxx86_64/hdb/manifest'
+HANA_SAPCONTROL_PATH = 'exe/linuxx86_64/hdb/sapcontrol'
+
+
+def perform_sap_hana_scan():
+    """
+    Produces a message with details collected around SAP HANA.
+    """
+
+    api.produce(search_sap_hana_instances())
+
+
+def parse_manifest(path):
+    """ Parses a SAP HANA manifest into a dictionary """
+    def _decoded(s):
+        if hasattr(s, 'decode'):
+            return s.decode('utf-8')
+        return s
+
+    data = {}
+    try:
+        with open(path, 'r') as f:
+            for line in _decoded(f.read()).split('\n'):
+                try:
+                    key, value = line.split(':', 1)
+                except ValueError:
+                    continue
```

Comment:
when this happens and why the continue is ok? a comment will be more clear.

Reply:
Comment added